src/cbortojson.c

```diff
@@ -146,8 +146,6 @@
  * the keys for the metadata clash with existing keys in the JSON map.
  */
 
-extern FILE *open_memstream(char **bufptr, size_t *sizeptr);
```

Comment:
This will break the build in systems where `open_memstream` isn't defined, that is, where we add src/open_memstream.c to the build. For example, like on macOS, which is part of our CI.

We need a new a way to inject this declaration into the build. The easiest is probably a -D flag next to this part of Makefile:
```make
ifeq ($(open_memstream-pass),)
  ifeq ($(funopen-pass)$(fopencookie-pass),)
    CFLAGS += -DWITHOUT_OPEN_MEMSTREAM
    ifeq ($(wildcard .config),.config)
        $(warning warning: funopen and fopencookie unavailable, open_memstream can not be implemented and conversion to JSON will not work prope
rly!)
    endif
  else
    TINYCBOR_SOURCES += src/open_memstream.c
  endif
endif
endif
```